snuba/subscriptions/executor_consumer.py

```diff
@@ -90,3 +102,127 @@ def create(
         self, commit: Callable[[Mapping[Partition, Position]], None]
     ) -> ProcessingStrategy[KafkaPayload]:
         return Noop(commit)
+
+
+class SubscriptionResultFuture(NamedTuple):
+    message: Message[KafkaPayload]
+    future: Future[Tuple[Request, Result]]
+
+
+class ExecuteQuery(ProcessingStrategy[KafkaPayload]):
+    """
+    Decodes a scheduled subscription task from the Kafka payload, builds
+    the request and executes the ClickHouse query.
+    """
+
+    def __init__(
+        self,
+        dataset: Dataset,
+        max_concurrent_queries: int,
+        metrics: MetricsBackend,
+        next_step: ProcessingStrategy[KafkaPayload],
+    ) -> None:
+        self.__dataset = dataset
+        self.__max_concurrent_queries = max_concurrent_queries
+        self.__metrics = metrics
+        self.__next_step = next_step
+
+        self.__encoder = SubscriptionScheduledTaskEncoder()
+
+        self.__executor = ThreadPoolExecutor()
```

Comment:
You should be able to create the executor at module level. Having it in the constructor means it will be recreated every time the strategy is recreated (including all rebalancing).